src/main/java/com/spotify/reaper/storage/CassandraStorage.java

```diff
@@ -707,26 +720,33 @@ public boolean updateRepairSchedule(RepairSchedule newRepairSchedule) {
   }
 
   public long getNewRepairId(String idType){
-    long idValue = 0;
-
-    // Create id counter if it doesn't exist yet
-    session.execute("INSERT INTO repair_id (id_type, id) VALUES('" + idType + "', 0) IF NOT EXISTS");
+    if (!repairIds.containsKey(idType)){
+        repairIds.putIfAbsent(idType, 0L);
+        // Create id counter if it doesn't exist yet
+        session.execute(insertRepairId.bind(idType));
+    }
+    long idValue = repairIds.get(idType);
+    int attempts = 0;
 
-    // Get current value of the counter, increment and perform CAS
+    // Increment and perform CAS, if it fails then fetch current value of the counter and repeat
     while(true){
-      idValue = session.execute("SELECT id FROM repair_id WHERE id_type = '" + idType + "'").one().getLong("id");
       idValue++;
-      ResultSet casResult = session.execute("UPDATE repair_id SET id=" + idValue + " WHERE id_type = '" + idType + "' IF id = " + (idValue-1));
+      ResultSet casResult = session.execute(updateRepairId.bind(idValue, idType, (idValue-1)));
       if(casResult.wasApplied()){
-        break;
+          break;
+      }else{
+          idValue = session.execute(selectRepairId.bind(idType)).one().getLong("id");
+          Preconditions.checkState(idValue < Long.MAX_VALUE);
+          attempts++;
+          if(10 <= attempts && 0 == attempts % 10){
+              LOG.warn("still cant find a new repairId after " + attempts + " attempts");
```

Comment:
of little concern. WARN is generally always enabled, and the frequency of this log line low.